docs/install-components/karmada.md

```diff
@@ -0,0 +1,163 @@
+
+## Prerequisites
+
+Installing Karmada requires a kubernetes cluster. If you do not have a kubernetes cluster, you can choose any of the following methods to deploy the kubernetes cluster.
+
+### Deploy kubernetes clusters using kurator's script
+
+Deploy a kubernetes cluster using kurator's scripts. This script will create three clusters for you, one is used to host Karmada control plane and the other two will be joined as member clusters.
+```bash
+git clone https://github.com/kurator-dev/kurator.git
+cd kurator
+hack/local-dev-setup.sh
+```
+
+### KinD
+
+Install docker
+
+[Install on Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+
+
+Install KinD
+```bash
+KIND_VERSION=v0.11.1
+curl -Lo ./kind "https://kind.sigs.k8s.io/dl/${KIND_VERSION}/kind-$(uname)-amd64"
+chmod +x ./kind
+mv ./kind /usr/local/bin/kind
+```
+
+Create a kubernetes cluster
+```bash
+kind create cluster
+```
+
+For more instructions, please refer to [KinD](https://kind.sigs.k8s.io/docs/user/quick-start)
+
+### kubeadm
+
+Load Linux kernel modules and set kernel parameters.
+```bash
+cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
+overlay
+br_netfilter
+EOF
+
+sudo modprobe overlay
+sudo modprobe br_netfilter
+
+# sysctl params required by setup, params persist across reboots
+cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
+net.bridge.bridge-nf-call-iptables  = 1
+net.bridge.bridge-nf-call-ip6tables = 1
+net.ipv4.ip_forward                 = 1
+EOF
+
+# Apply sysctl params without reboot
+sudo sysctl --system
+```
+
+Install containerd
+```
+sudo apt-get install  containerd
+```
+
+Install Kubeadm Kubelet Kubectl
+```bash
+sudo apt-get update && sudo apt-get install -y apt-transport-https curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+cat <<EOF | sudo tee /etc/apt/sources.list.d/kubernetes.list
+deb https://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+
+sudo apt-get update
+sudo apt-get install -y kubelet kubeadm kubectl
+sudo apt-mark hold kubelet kubeadm kubectl
+```
+
+Create a kubernetes cluster
+```
+kubeadm config images pull
+kubeadm init
+
+mkdir -p $HOME/.kube
+sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u):$(id -g) $HOME/.kube/config
+```
+
+Install CNI, take calico as an example
+```
+curl https://docs.projectcalico.org/manifests/calico.yaml -O
+kubectl apply -f calico.yaml
+kubectl -n kube-system get pod -w
+```
+> Notice
+> 
+> In the kubernetes cluster created by Kubeadm, the master node is identified as Taints. If your kubernetes cluster has only one node, you need to remove the Taints of the master node.
+> 
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/control-plane-`
+> 
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/master-`
+
+
+For more instructions, please refer to [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm)
+
+## Deploy Karmada
+
+Compile `kurator` from source
```

Comment:
maybe we should link this to the README.md of repo, which I think will keep updating.

another option is I will take a look about the doc infra, want to find a way that we can insert snippet like istio.io do.

Reply:
Sorry, I don't quite understand what you mean.

Personally I think the repo's README.md is quickstart, and the installation documentation should be more detailed than quickstart.

Reply:
However, about compiling kurator from source or downloading kurator, it really shouldn't appear here.

Reply:
After thinking, let's put this section here at this stage.